### DISCUSSION TITLE
「开源之史」系列之一：该如何思考开源的历史？ - 开源之道

### DISCUSSION THREAD

**Discussion by giscus** [OCselected/ocselected.github.io#9] [Announcements]
# 「开源之史」系列之一：该如何思考开源的历史？ - 开源之道

https://opensourceway.community/posts/history-of-open-source/00-how-to-think-in-history-view/

**Comment by SarahSmj** [comment]
李先生您好，我是秒如科技的HR，是从拉勾上看到您的博客找过来的，我们诚聘开源布道师一职，若您感兴趣，可以添加微信：smj1989，可以进一步跟您沟通，期待您回复

**Comment by ZoomQuiet** [comment]
#是也乎,(￣▽￣)

> ...根本就不存在什么开源软件供应链

哈, 这个点, 可以这么理解:

- 开源的供应链, 虽然有, 但, 却是其它商业机构无法插足的
- 其供应链接太短了, 只有想法到作品一个单一环节, 依靠一个或是一组热情/有能力/自律/礼貌的创造人完成连接

也就是说, 自由/开源软件本身就是自给自足的, 除了必要的电力/CocaCola/以及激情的输入, 就可以生产出一切;
当然, 软件是要运行在硬件上的,
自由/开源世界所运行的硬件世界以前也是开源的, 
后来即便是闭源了, 但是为了确保用户可以安全/合理使用, 硬件的使用手册是开源的, 允许自由传播和改进的...

所以, 可能进一步的认为, 这个世界上根本就不存在开源以外的软件供应链, 
(当然, 这种从硬件到软件全部是封闭专用的系统也有, 比如一些特殊的军用装备系统, 但是, 这种系统也不为外界接触/使用, 对于人类社会也等于不存在了...)
没有它者, 自然也就没有链 之说了;

**Reply by lijiangsheng1** [reply]
同意，一个人的世界就确实啥都不存在了，可人类世界的神奇的地方就在于没有人是一座孤岛。

**Comment by zhuangbiaowei** [comment]
打算认真读一遍这个系列，然后将读后感评论在各篇文章下面。

1. 关于时间轴，我觉得虽然时间轴可能会给人一种“上升趋势”的感觉，但是历史事件一定是发生在时间里的，将多种时间分别排列在时间序列之中，然后既考虑先后发生的事件的纵向影响，也考虑同时发生的事件的横向关联。我觉得是比较合理的思路。
2. 关于供应链，我的判断是：肯定存在一款开源软件，里面全部都是开源代码。而很少能够找到商业闭源软件，里面全是闭源自研的代码。因此：纯粹的开源供应链，还是存在的。
3. 关于开源历史的写法，我非常赞同适兕的思路，是需要将嵌入在整个大历史中的开源，依各种各样的线索去分析与理解的。这种思考历史的方式，才能对开源做最全面的理解。

**Reply by tisonkun** [reply]
> 肯定存在一款开源软件，里面全部都是开源代码。

很多呀，Apache 软件基金会的作品都会要求只能包含开源代码。

* See also https://www.apache.org/legal/resolved.html

**Reply by lijiangsheng1** [reply]
Debian 就是最纯粹的～ GNU/Linux 认证的也是～ :-)
